c/driver_manager/adbc_driver_manager.cc

```diff
@@ -632,6 +985,36 @@ static const char kDefaultEntrypoint[] = "AdbcDriverInit";
 
 // Other helpers (intentionally not in an anonymous namespace so they can be tested)
 
+std::vector<std::filesystem::path> AdbcParsePath(const std::string_view& path) {
+  std::vector<std::filesystem::path> result;
+  if (path.empty()) {
+    return result;
+  }
+
+#ifdef _WIN32
+  constexpr char delimiter = ';';
+#else
+  constexpr char delimiter = ':';
+#endif  // _WIN32
+
+  bool in_quotes = false;
```

Comment:
Hmm, is escaping actually a thing? https://unix.stackexchange.com/questions/579081/why-assign-export-ld-library-path suggests it conventionally isn't. (Also it notes that Linux can use `;` too.)

Reply:
The escaping is needed if there are any spaces in the path. Much more common on Windows than on Linux, but still possible.

I was following the same logic as Go's os package for parsing path delimiters (: for Linux, ; for Windows)

Reply:
Er, I don't understand. Escaping spaces is the responsibility of the shell, not us

Reply:
The other reason for needing the quote handling is if you have a path which contains the path separator (: or ;) in the folder itself.

Reply:
Right, but apparently conventionally this isn't handled, so why are we inventing a convention? (Also how do you escape quotes? And is there any precedent for doing this vs a backslash?)

Reply:
Go only does that on Windows, though

Reply:
I can't find a good reference but it seems this is implicit/expected behavior on Windows, but isn't on Linux. If we're going to do this we should at least point to the reference being used for the behavior

Reply:
I assume the goal was to follow platform conventions in the first place. That includes not bleeding over behavior from one platform to another

Reply:
Sure, I'll update this then and add a comment linking to the reference. 

There's surprisingly few examples of prior art here that I can find

Reply:
I also expect others may end up being more opinionated on this than us.... Lol

Reply:
In other discussions we've come to the conclusion to avoid exposing a version that takes the list of strings. I'll update this to work more similar to https://cs.opensource.google/go/go/+/refs/tags/go1.24.3:src/path/filepath/path_windows.go (only do the quotes/escaping check on windows, etc.)